tests/testthat/test_exponential_decay_corr_func.R

```diff
@@ -0,0 +1,38 @@
+test_that(
+  "Test Exponential Decay Corr Func in Stan agrees with R",
+  {
+    model <- compiled_site_inf_model
+    space_model_fxns <- spatial_fxns
+
+    corr_func_params <- list(
```

Comment:
set the seed here for reproducibility? also, I'd suggest using `withr::with_seed` rather than just calling `set.seed` within the individual tests https://withr.r-lib.org/reference/with_seed.html